### PR TITLE
Updating website workflow

### PR DESCRIPTION
**Pull request by naskio**
In order to simplify the processing of Dependabot PRs, I suggest having the following behaviors based on the current branch:

1. Build and deploy if on the `main` branch.
2. Test if the website builds successfully if on a `dependabot` branch.

### CODE REVIEW

**Review by suprjinx — APPROVED** [review]

**Comment by jgiannuzzi** [comment]
@naskio if the goal is this PR is to validate that any changes to the website files are not breaking the website itself, I would suggest building the website on pull requests and pushes to any branch, so that other kinds of changes can be validated too

File: .github/workflows/website.yml [hunk 1]
```diff
@@ -1,9 +1,7 @@
-name: Deploy website to Pages
+name: Build and Deploy website to Pages
 
 on:
-  push: # Runs on pushes targeting the default branch
-    branches:
-      - main
+  push: # Runs on pushes targeting any branch
```
Comment (jgiannuzzi) [inline .github/workflows/website.yml:4]: we also need to check for pull requests — have a look at how it's done in `ci.yml`

Comment (naskio) [reply .github/workflows/website.yml:4]: Thanks, should be fixed now

Comment (naskio) [reply .github/workflows/website.yml:4]: I was just wondering if there are cases not covered by on `push` and covered by on `pull_request`, maybe when doing changes using GitHub editor ?

Comment (naskio) [reply .github/workflows/website.yml:4]: Ah forgot about forks xD never mind.

**Review by jgiannuzzi — CHANGES_REQUESTED** [review]
one more thing: make sure we run only once for either PR or push to repo branch — again, have a look at how we do it in `ci.yml`!

**Comment by naskio** [comment]
I tried to use `concurrency.group` instead of `if`, since it was already being used. Let me know if you prefer to do it otherwise.

**Review by jgiannuzzi — APPROVED** [review]
LGTM, thanks @naskio!

**Pull request merged**